src/System.Reflection.Emit/tests/DynamicMethodJumpStubTests.cs

```diff
@@ -0,0 +1,126 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using System.Threading;
+using Xunit;
+
+namespace System.Reflection.Emit.Tests
+{
+    public static class DynamicMethodJumpStubTests
+    {
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public static void DynamicMethodJumpStubTest()
+        {
+            if (RuntimeInformation.ProcessArchitecture != Architecture.X64)
+            {
+                return;
+            }
+
+            // Reserve memory around framework libraries. This is just a best attempt, it typically doesn't help since the
+            // precode allocator may have already committed pages it can allocate from, or it may commit reserved pages close to
+            // framework libraries.
+            ReserveMemoryAround(new Action(ExecutionContext.RestoreFlow).Method.MethodHandle);
+
+            for (int i = 0; i < 64; ++i)
+            {
+                DynamicMethod dynamicMethod = CreateDynamicMethod("DynMethod" + i);
+                Action dynamicMethodDelegate = (Action)dynamicMethod.CreateDelegate(typeof(Action));
+
+                // Before compiling the dynamic method, reserve memory around its current entry point, which should be its
+                // precode. Then, when compiling the method, there would be a good chance that the code will be located far from
+                // the precode, forcing the use of a jump stub.
+                ReserveMemoryAround(
+                    (RuntimeMethodHandle)
+                    typeof(DynamicMethod).InvokeMember(
+                        "GetMethodDescriptor",
+                        BindingFlags.Instance | BindingFlags.InvokeMethod | BindingFlags.NonPublic,
+                        null,
+                        dynamicMethod,
+                        null));
+
+                dynamicMethodDelegate();
+            }
+
+            // This test does not release reserved pages because they may have been committed by other components on the system
+        }
+
+        private static DynamicMethod CreateDynamicMethod(string name)
+        {
+            var dynamicMethod = new DynamicMethod(name, null, null);
+            ILGenerator ilGenerator = dynamicMethod.GetILGenerator();
+            ilGenerator.Emit(OpCodes.Ret);
+            return dynamicMethod;
+        }
+
+        private const uint AllocationGranularity = (uint)64 << 10;
+        private const ulong ReserveRangeRadius = (ulong)4 << 30; // reserve 4 GB before and after the base address
```

Comment:
I wanted to allocate more to cover edge cases. Also for reserving memory around framework libraries, I only have one entry point to work with, so the idea was that hopefully the additional 2 GB on each side will cover the rest of the libraries as well.